editor/src/clj/editor/tile_map.clj

```diff
@@ -1207,20 +1242,29 @@
   (let [^Point3d screen-pos (:screen-pos action)]
     (case (:type action)
       :mouse-pressed
-      true
+      (do
+        (g/transact
+          (g/set-property self :start-palette-tile (g/node-value self :palette-tile evaluation-context)))
+        true)
 
       :mouse-moved
-      (g/transact
-        (g/set-property self :cursor-screen-pos screen-pos))
+      (do
+        (g/transact
+          (g/set-property self :cursor-screen-pos screen-pos))
+        true)
 
       :mouse-released
-      (let [palette-tile (g/node-value self :palette-tile evaluation-context)]
+      (let [start-tile (g/node-value self :start-palette-tile evaluation-context)
+            end-tile (g/node-value self :palette-tile evaluation-context)]
         (g/transact
           (concat
-            (g/set-property self :brush (make-brush palette-tile))
-            (g/set-property self :mode :editor))))
-
-      action)))
+            (when (and start-tile end-tile)
+              [(g/set-property self :brush
+                 (make-brush-from-selection-in-palette start-tile end-tile (g/node-value self :tile-source-attributes evaluation-context)))])
```

Comment:
This is fine, because any transaction steps that are inside of lists will be flattened, but this does not technically need to be inside a vector, since `g/set-property` actually returns a vector with a single transaction step in it. All the `g/*` functions that return transaction steps return vectors like this.

Anyway, I just wanted to mention it for future reference. It's fine to merge like this, really.